src/lib/homestore/heap_chunk_selector.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <homestore/chunk_selector.h>
+#include <homestore/vchunk.h>
+#include <homestore/homestore_decl.hpp>
+
+#include <queue>
+#include <vector>
+#include <mutex>
+#include <utility>
+#include <memory>
+
+#include <folly/concurrency/ConcurrentHashMap.h>
+
+namespace homeobject {
+
+using csharedChunk = homestore::cshared<homestore::Chunk>;  
+
+class HeapChunkSelector : public homestore::ChunkSelector {
+public:
+    HeapChunkSelector() = default;
+    HeapChunkSelector(const HeapChunkSelector&) = delete;
+    HeapChunkSelector(HeapChunkSelector&&) noexcept = delete;
```

Comment:
Why do we explicitly delete this? We have a default CTor and DTor; let the compiler decide if we can support this by default based on member fields?